test/test_model.py

```diff
@@ -796,6 +796,17 @@ def test_planned_units(self):
 
         self.assertEqual(self.app.planned_units(), 3)
 
+    def test_planned_units_user_set(self):
+
+        self.harness.set_planned_units(1)
+        self.assertEqual(self.app.planned_units(), 1)
+
+        self.harness.set_planned_units(2)
+        self.assertEqual(self.app.planned_units(), 2)
+
+        self.harness.set_planned_units(100)
+        self.assertEqual(self.app.planned_units(), 100)
+
```

Comment:
I feel like there should be some sort of test that handles the case of:
```
  self.harness.set_planned_units(1)
  self.harness.add_relation_unit(peer_id, 'myapp/1')
  self.harness.add_relation_unit(peer_id, 'myapp/2')
  self.harness.add_relation_unit(peer_id, 'myapp/3')
```

I'm not 100% sure what should happen there. Maybe it is ok, as it is modeling the case of 'remove-unit'. (you do currently have 3 units in scope,but someone did a 'juju remove-unit' on them and we should be treating them as going away.)
At the very least I would want us to have an explicit test of this behavior and a definition of how we expect it to work.

Reply:
I think that the correct behavior here is that planned_units is still 1 at the end of all this, because we want to respect what the test author intended. I went ahead and formalized that decision in a test.